pkg/storageincentives/agent.go

```diff
@@ -298,8 +294,26 @@ func (a *Agent) claim(ctx context.Context) error {
 	}
 
 	if isWinner {
+		a.nodeState.SetLastWonRound(round)
 		a.metrics.Winner.Inc()
-		err = a.contract.Claim(ctx)
+		errBalance := a.nodeState.SetBalance(ctx)
+		if errBalance != nil {
+			a.logger.Info("could not set balance", "err", err)
+		}
+
+		fee, err := a.contract.Claim(ctx)
+		if err != nil {
+			a.logger.Info("calculate winner reward", "err", err)
+			return err
+		}
+		if errBalance == nil {
+			err = a.nodeState.CalculateWinnerReward(ctx)
+			if err != nil {
+				a.logger.Info("calculate winner reward", "err", err)
+			}
+		}
+		a.nodeState.AddFee(fee)
+
 		if err != nil {
```

Comment:
bad left over code

Reply:
I don't get it ?

Reply:
which error is this checking ?

Reply:
Its for  `err := a.contract.Claim(ctx)` 
Anyway, I have changed calculate error name to remove confusion.

Reply:
you are already checking it on line 305 😭

Reply:
I'd recommend reviewing the code yourself once more

Reply:
I haven't added this check. I thought it might be for metrics. Anyway updated it.